### PR TITLE
Add default_tags option

### PR DESCRIPTION
**Pull request by mikker**
Allows the user to add config-level tags that'll be included in every transaction.

See elastic/apm-agent-ruby#164 

Also includes some spec suite clean-up.

### CODE REVIEW

File: bench/report.rb [hunk 1]
```diff
@@ -50,7 +50,7 @@
     transaction_count: count,
     executed_at: Time.new.iso8601,
     'git.commit' => git_sha,
-    'git.date' => git_date && Time.parse(git_date).iso8601,
+    'git.date' => String(git_date).strip != '' && Time.parse(git_date).iso8601,
```
Comment (simitt) [inline bench/report.rb]: Could you move this to a separate commit/PR please?

Comment (mikker) [reply bench/report.rb]: Too late 😬

File: lib/elastic_apm/transaction.rb [hunk 2]
```diff
@@ -29,6 +29,7 @@ def initialize(
       @notifications = [] # for AS::Notifications
 
       @context = context || Context.new
+      @context.tags.merge!(instrumenter.config.default_tags)
```
[the remark this answers is not in the document]
Comment (mikker) [reply lib/elastic_apm/transaction.rb]: It probably should, yes 👍
